scipy/sparse/dok.py

```diff
@@ -290,21 +291,19 @@ def __setitem__(self, index, x):
                     del self[key]
 
     def __add__(self, other):
-        # First check if argument is a scalar
-        if isscalarlike(other):
+        if isscalarlike(other):  # check if argument is a scalar
             res_dtype = upcast_scalar(self.dtype, other)
             new = dok_matrix(self.shape, dtype=res_dtype)
             # Add this scalar to every element.
             M, N = self.shape
-            for i in xrange(M):
-                for j in xrange(N):
-                    aij = self.get((i, j), 0) + other
-                    if aij != 0:
-                        new[i, j] = aij
+            for i, j in itertools.product(xrange(M), xrange(N)):
+                aij = dict.get(self, (i, j), 0) + other
+                if aij:
+                    new[i, j] = aij
             # new.dtype.char = self.dtype.char
-        elif isinstance(other, dok_matrix):
+        elif isspmatrix_dok(other):
```

Comment:
To me its also not any clearer but there is a function isspmatrix_dok that, I guess, was supposed to be used for this otherwise it has no purpose?